runtime/vm/ContinuationHelpers.cpp

```diff
@@ -94,23 +94,32 @@ synchronizeWithConcurrentGCScan(J9VMThread *currentThread, J9VMContinuation *con
 {
 	volatile uintptr_t *localAddr = &continuation->state;
 	/* atomically 'or' (not 'set') continuation->state  with currentThread */
-	uintptr_t oldContinuationState = VM_AtomicSupport::bitOr(localAddr, (uintptr_t)currentThread);
+	PORT_ACCESS_FROM_VMC(currentThread);
+	j9tty_printf(PORTLIB, "synchronizeWithConcurrentGCScan before continuation=%p, continuation->state=%p, currentThread=%p\n", continuation, *localAddr, currentThread);
+	uintptr_t oldContinuationState = VM_AtomicSupport::bitOr(localAddr, (uintptr_t)currentThread | VM_VMHelpers::getPendingToBeMountedMask());
+	j9tty_printf(PORTLIB, "synchronizeWithConcurrentGCScan bitOr continuation=%p, continuation->state=%p, oldContinuationState=%p, currentThread=%p\n", continuation, *localAddr, oldContinuationState, currentThread);
 
 	Assert_VM_Null(VM_VMHelpers::getCarrierThreadFromContinuationState(oldContinuationState));
 
-	if (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(oldContinuationState)) {
-		/* currentThread was low tagged (GC was already in progress), but by 'or'-ing our ID, we let GC know there is a pending mount */
-		internalReleaseVMAccess(currentThread);
+	do {
+		if (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(oldContinuationState)) {
+			/* currentThread was low tagged (GC was already in progress), but by 'or'-ing our ID, we let GC know there is a pending mount */
+			j9tty_printf(PORTLIB, "synchronizeWithConcurrentGCScan pending to be mounted continuation=%p, continuation->state=%p, oldContinuationState=%p, currentThread=%p\n", continuation, *localAddr, oldContinuationState, currentThread);
+			internalReleaseVMAccess(currentThread);
+
+			omrthread_monitor_enter(currentThread->publicFlagsMutex);
+			while (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(*localAddr)) {
+				/* GC is still concurrently scanning the continuation(currentThread was still low tagged), wait for GC thread to notify us when it's done. */
+				omrthread_monitor_wait(currentThread->publicFlagsMutex);
+			}
+			omrthread_monitor_exit(currentThread->publicFlagsMutex);
 
-		omrthread_monitor_enter(currentThread->publicFlagsMutex);
-		while (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(*localAddr)) {
-			/* GC is still concurrently scanning the continuation(currentThread was still low tagged), wait for GC thread to notify us when it's done. */
-			omrthread_monitor_wait(currentThread->publicFlagsMutex);
+			internalAcquireVMAccess(currentThread);
 		}
-		omrthread_monitor_exit(currentThread->publicFlagsMutex);
+		oldContinuationState = VM_AtomicSupport::lockCompareExchange(localAddr, (uintptr_t)currentThread | VM_VMHelpers::getPendingToBeMountedMask(), (uintptr_t)currentThread);
```

Comment:
have to think about this more, but I have a feeling this should be set together with carrierID early, and removed once we get out of this sync step